client/components/button-group/test/index.js

```diff
@@ -1,25 +1,27 @@
+/**
+ * @jest-environment jsdom
+ */
+import '@testing-library/jest-dom';
 import { Button } from '@automattic/components';
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 import ButtonGroup from '..';
 
 describe( 'ButtonGroup', () => {
 	test( 'should have ButtonGroup class', () => {
-		const buttonGroup = shallow( <ButtonGroup /> );
-		expect( buttonGroup.find( '.button-group' ) ).toHaveLength( 1 );
+		render( <ButtonGroup /> );
+		expect( screen.getByTestId( 'button-group-id' ) ).toHaveClass( 'button-group' );
 	} );
-
 	test( 'should contains the same number of .button nodes than <Button>s it receives', () => {
-		const buttonGroup = shallow(
+		render(
 			<ButtonGroup>
 				<Button>test</Button>
 				<Button>test2</Button>
 			</ButtonGroup>
 		);
-		expect( buttonGroup.find( Button ) ).toHaveLength( 2 );
+		expect( screen.getAllByRole( 'button' ).length ).toBe( 2 );
```

Comment:
Nice that we're using `getAllByRole` 👏 

We could improve a little bit by using the idiomatic `toHaveLength()`:

```suggestion
		expect( screen.getAllByRole( 'button' ) ).toHaveLength( 2 );
```